src/hermes/commands/deposit/invenio.py

```diff
@@ -484,3 +488,70 @@ def _get_community_identifiers(ctx: CodeMetaContext, communities_api_url: str):
         community_ids.append({"identifier": response.json()["id"]})
 
     return community_ids
+
+
+def _get_access_modalities(license):
+    """Get access right, embargo date and access conditions based on configuration and given license.
+
+    This function implements the rules laid out in the `Zenodo developer documentation
+    <https://developers.zenodo.org/#representation>`_:
+
+    - ``access_right`` is a controlled vocabulary
+    - embargoed access depositions need an embargo date
+    - restricted access depositions need access conditions
+    - open and embargoed access depositions need a license
+    - closed access depositions have no further requirements
+
+    This function also makes sure that the given embargo date can be parsed as an ISO
+    8061 string representation and that the access rights are given as a string.
+    """
+    invenio_config = config.get("deposit").get("invenio", {})
+
+    access_right = invenio_config.get("access_right")
+    if access_right is None:
+        raise MisconfigurationError("deposit.invenio.access_right is not configured")
+
+    access_right_options = ["open", "embargoed", "restricted", "closed"]
+    if access_right not in access_right_options:
+        raise MisconfigurationError(
+            "deposition.invenio.access_right must be one of: "
+            f"{', '.join(access_right_options)}"
+        )
+
+    embargo_date = invenio_config.get("embargo_date")
+    if access_right == "embargoed" and embargo_date is None:
+        raise MisconfigurationError(
+            f"With access_right {access_right}, "
+            "deposit.invenio.embargo_date must be configured"
+        )
+
+    if embargo_date is not None:
+        try:
+            datetime.fromisoformat(embargo_date)
+        except ValueError:
+            raise MisconfigurationError(
+                f"Could not parse deposit.invenio.embargo_date {embargo_date!r}. "
+                "Must be in ISO 8601 format."
+            )
+
+    access_conditions = invenio_config.get("access_conditions")
+    if access_right == "restricted" and access_conditions is None:
+        raise MisconfigurationError(
+            f"With access_right {access_right}, "
+            "deposit.invenio.access_conditions must be configured"
+        )
+
+    if access_conditions is not None and not isinstance(access_conditions, str):
+        raise MisconfigurationError(
+            "deposit.invenio.access_conditions must be a string (HTML is allowed)."
+        )
+
+    if license is None and access_right in ["open", "embargoed"]:
+        raise MisconfigurationError(
+            f"With access_right {access_right}, a license is required."
+        )
+
+    if access_right == "closed":
+        pass
+
+    return access_right, embargo_date, access_conditions
```

Comment:
*note:* I see lots of potential for abstraction here... we have a number of dependent fields, each is checked for being required and if there is a valid value given.
*conclusion:* We should do a proper code review after PoC and collect our findings